hudi-io/src/main/java/org/apache/hudi/storage/HoodieStorage.java

```diff
@@ -370,7 +370,7 @@ public final void createImmutableFileInPath(StoragePath path,
    */
   @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
   public final boolean needCreateTempFile() {
-    return StorageSchemes.HDFS.getScheme().equals(getScheme());
+    return StorageSchemes.HDFS.getScheme().equals(getScheme()) || StorageSchemes.FILE.getScheme().equals(getScheme());
```

Comment:
We still need this judgement to pass our unit-tests.

Reply:
Can we revert this change and just fix the unit tests.

Reply:
> Can we revert this change and just fix the unit tests.

OK, I will just fix ut but not change this method~

Reply:
> Can we revert this change and just fix the unit tests.

done~